src/main/java/org/openstreetmap/atlas/checks/validation/points/InvalidMiniRoundaboutCheck.java

```diff
@@ -0,0 +1,149 @@
+package org.openstreetmap.atlas.checks.validation.points;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import org.openstreetmap.atlas.checks.base.BaseCheck;
+import org.openstreetmap.atlas.checks.flag.CheckFlag;
+import org.openstreetmap.atlas.geography.atlas.items.AtlasObject;
+import org.openstreetmap.atlas.geography.atlas.items.Edge;
+import org.openstreetmap.atlas.geography.atlas.items.Node;
+import org.openstreetmap.atlas.tags.DirectionTag;
+import org.openstreetmap.atlas.tags.HighwayTag;
+import org.openstreetmap.atlas.tags.Taggable;
+import org.openstreetmap.atlas.tags.annotations.validation.Validators;
+import org.openstreetmap.atlas.utilities.collections.Iterables;
+import org.openstreetmap.atlas.utilities.configuration.Configuration;
+
+/**
+ * Flags any Nodes tagged as highway=MINI_ROUNDABOUT that do not have very many incoming/outgoing
+ * Edges as potentially tagged incorrectly.
+ *
+ * @author nachtm
+ */
+public class InvalidMiniRoundaboutCheck extends BaseCheck<Long>
+{
+
+    private static final long DEFAULT_VALENCE = 6;
+    private static final String MINIMUM_VALENCE_KEY = "valence.minimum";
+    private static final String OTHER_EDGES_INSTRUCTION = "This Mini-Roundabout Node ({0,number,#})"
```

Comment:
Nit: It might make sense to make the instructions in terms of Ways instead of Edges. The average user won't understand how an OSM Way maps to a car-navigable edge.